### ISSUE TITLE
Question about training time.

### ISSUE THREAD

**Issue by muuda** [dberghi/AV-SELD#3]
Hello, thank you very much for open-sourcing your code. I followed your instructions and found that training one epoch takes about 8 hours. Is this normal? My GPU is an NVIDIA GeForce RTX 3090 24G, and my batch size is 32.

**Comment by dberghi** [comment]
Hello and thank you for your interest in our work.

I recall that each epoch takes a few hours, but 8 seems a bit long. I'll need to run a test myself to provide a more accurate estimate. Which audio-visual fusion method are you training? AV-conformer or CMAF? 

In any case, don't train for 50 epochs as mentioned in the paper, 10 epochs should be enough. From what I remember, it was taking less than 2 days to complete 10 epochs. I will run the training on a 3090 and get back to you.

In the meantime, you should be able to check the progress of your training from `./output/<INFO>/log_<LR>.txt` where the test set results are printed

**Comment by muuda** [comment]
Thank you for your reply. This is very helpful for my work.

My training parameters are as follows
``` 
training_param = {
    'optimizer': torch.optim.Adam,
    #'criterion': nn.CrossEntropyLoss,
    'learning_rate': 0.00005, # default if user does not provide a different lr to the parser
    'epochs': 10, # default if user does not provide a different number to the parser
    'batch_size': 32, # default if user does not provide a different size to the parser
    'frame_len_samples': input['input_len_sec'] * input['fs'], # number of audio samples in input_len_sec,
    'num_video_frames': input['input_len_sec'] * input['fps'], # number of video frames in input_len_sec,
    'visual_encoder_type': 'resnet', # choose between 'resnet' or 'i3d'
    'model_type': 'conformer', # choose between 'conformer' or 'cmaf'
    'num_heads': 8, # number of heads in MHSA-MHCA
    'num_cmaf_layers': 4,
}
```

I checked the `./output/<INFO>/log_<LR>.txt` file, where the test set results were successfully printed.
